client/rest-high-level/src/test/java/org/elasticsearch/client/RequestTests.java

```diff
@@ -428,6 +430,49 @@ public void testDeleteIndex() {
         assertNull(request.getEntity());
     }
 
+    public void testGetSettings() throws IOException {
+        String[] indicesUnderTest = randomBoolean() ? null : randomIndicesNames(0, 5);
+
+        GetSettingsRequest getSettingsRequest = new GetSettingsRequest().indices(indicesUnderTest);
+
+        Map<String, String> expectedParams = new HashMap<>();
+        setRandomMasterTimeout(getSettingsRequest, expectedParams);
+        setRandomIndicesOptions(getSettingsRequest::indicesOptions, getSettingsRequest::indicesOptions, expectedParams);
+
+        setRandomLocal(getSettingsRequest, expectedParams);
+
+        if (randomBoolean()) {
+            //the request object will not have include_defaults present unless it is set to true
+            getSettingsRequest.includeDefaults(true);
+            expectedParams.put("include_defaults", Boolean.toString(true));
+        }
```

Comment:
nit: could you also randomly set it to the request when false, just to make sure? Maybe something like:

```
if (randomBoolean()) {
    getSettingsRequest.includeDefaults(randomBoolean());
    if (getSettingsRequest.includeDefaults()) {
        expectedParams.put("include_defaults", Boolean.toString(true));
    }
}
```